types/src/transaction/authenticator.rs

```diff
@@ -1027,22 +1058,86 @@ impl AnyPublicKey {
         Self::Secp256r1Ecdsa { public_key }
     }
 
+    pub fn zkid(public_key: ZkIdPublicKey) -> Self {
+        Self::ZkId { public_key }
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        bcs::to_bytes(self).expect("Only unhandleable errors happen here.")
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum EphemeralSignature {
+    Ed25519 { signature: Ed25519Signature },
+}
+
+impl EphemeralSignature {
+    pub fn ed25519(signature: Ed25519Signature) -> Self {
+        Self::Ed25519 { signature }
+    }
+
+    pub fn verify<T: Serialize + CryptoHash>(
+        &self,
+        message: &T,
+        public_key: &EphemeralPublicKey,
+    ) -> Result<()> {
+        match (self, public_key) {
+            (Self::Ed25519 { signature }, EphemeralPublicKey::Ed25519 { public_key }) => {
+                signature.verify(message, public_key)
+            },
```

Comment:
Nit: add `_ => {}` for explicitness

Reply:
I originally had that but cargo check gives me 

```
unreachable pattern
`#[warn(unreachable_patterns)]` on by default
```